### ISSUE TITLE
Bitbucket API and Typescript Type Mismatch  

### ISSUE THREAD

**Issue by forcebewithyou** [MunifTanjim/node-bitbucket#128]
I am fairly new to node and typescript, so it might very well be a developer error :) 

Issue am facing is that I am forced to include "type" as a required attribute in my requests which bitbucket api is complaining. 

I believe its due to "object" being included as base extension and definition of "object" has "type" as required attribute. 

<img width="507" alt="image" src="https://user-images.githubusercontent.com/1187274/189246287-fd220e24-d2e2-4f99-ba91-50acc414b8c3.png">

And interface object has type as required attribute 

<img width="334" alt="image" src="https://user-images.githubusercontent.com/1187274/189246348-7b705305-c2b1-4994-bab2-0491fb383932.png">

**Comment by MunifTanjim** [comment]
The types are auto generated from BitBucket's API Specification:
- https://api.bitbucket.org/swagger.json
- https://github.com/MunifTanjim/node-bitbucket/blob/dba6ff4/scripts/generate-types.js

What problem are you facing, btw?

**Comment by forcebewithyou** [comment]
Am trying to post a comment to PR inline 
<img width="594" alt="image" src="https://user-images.githubusercontent.com/1187274/189355599-7cc4244d-530e-41fb-b241-73d9e01ee2f2.png">

Typescript is complaining that "type" is a required attribute. 

error TS2322: Type '{ content: { raw: string; }; inline: { path: string; to: number; }; }' is not assignable to type 'PullrequestComment'.
  Property 'type' is missing in type '{ content: { raw: string; }; inline: { path: string; to: number; }; }' but required in type 'Object'.

Now when I add type attribute 
<img width="636" alt="image" src="https://user-images.githubusercontent.com/1187274/189355925-dc098122-a9f8-4b07-99b5-53435fa07658.png">

Bitbucket API complain bad request. 

When I update schema and make Type as optional and send the request without it, voila.. it works.

**Comment by MunifTanjim** [comment]
Yeah, those specification are definitely wrong. [BitBucket's API specification](https://api.bitbucket.org/swagger.json) has way too many incorrect info.

Since this whole library is auto-generated from those specification, it's affected by it. It's very hard for me to know which part of the speification is correct and which part is wrong, so it's not possible for me to patch definitions for each endpoint.

Unless BitBucket fixes those, I'm afraid I can't be of much help here.

**Comment by forcebewithyou** [comment]
I understand, thank you for prompt responses, went ahead with direct integration using node-fetch.
